pkg/plugin/test-keeper/plugin/test_checker.go

```diff
@@ -12,23 +12,48 @@ type TestChecker struct {
 	TestKeeperMatcher TestMatcher
 }
 
-// IsAnyNotExcludedFileTest checks if a commit affects any test file
-func (checker *TestChecker) IsAnyNotExcludedFileTest(files []scm.ChangedFile) (bool, error) {
+// FileCategories holds information about the total files coming in the changeset, legit files (those which are excluded from test verification)
+// and Tests
+type FileCategories struct {
+	Total, Legit, Tests int
+	Files               *[]scm.ChangedFile
+}
+
+// OnlyLegitFiles indicates if changeset contains only files which are excluded from test verification
+func (f *FileCategories) OnlyLegitFiles() bool {
+	return f.Total > 0 && f.Legit == f.Total
+}
+
+// TestsExist answers if any test files are found
+func (f *FileCategories) TestsExist() bool {
+	return f.Tests > 0
+}
+
+// NewFileCategories creates new instance of FileCategories struct with files populated
+func NewFileCategories(files []scm.ChangedFile) FileCategories {
+	return FileCategories{Files: &files, Total: len(files)}
+}
 
-	remainingNoTestFiles := false
+// CategorizeFiles counts files in the changeset which are tests (included files) and should not be considered for
+// verification (excluded). When first test is found it stops, as this is enough to unblock PR
+func (checker *TestChecker) CategorizeFiles(files []scm.ChangedFile) (FileCategories, error) {
```

Comment:
Is this still `TestChecker` or something else? 
btw, when I read `CategorizeFiles` I would expect that I get some categories - slices of files separated into different categories, not only count of the file types. Would make sense to rename it?:
`TestChecker` -> `FileTypeCounter`
`CategorizeFiles` -> `Count`
`FileCategories` -> `FileTypes` or `CountedFileTypes`

I know that I'm not good at naming stuff, my proposal is maybe worse than yours, it just didn't make sense to me at the very first look.

Reply:
That's right, maybe `Categorize` is too much for just counting them up. Let me reword it.

Reply:
e9377ce